src/nemos/basis/_basis.py

```diff
@@ -0,0 +1,1406 @@
+        start_slice :
+            The starting index for slicing, by default it starts from 0.
+        split_by_input :
+            Flag indicating whether to split the slicing by individual inputs or not.
+            If ``False``, a single slice is generated for all inputs.
+
+        Returns
+        -------
+        split_dict :
+            Dictionary with keys as labels and values as slices representing
+            the slicing for each input or additive component, if split_by_input equals to
+            True or False respectively.
+        start_slice :
+            The updated starting index after slicing.
+
+        See Also
+        --------
+        _get_default_slicing : Handles default slicing logic.
+        _merge_slicing_dicts : Merges multiple slicing dictionaries, handling keys conflicts.
+        """
+        # Set default values for n_inputs and start_slice if not provided
+        n_inputs = n_inputs or self._n_basis_input
+        start_slice = start_slice or 0
+
+        # If the instance is of AdditiveBasis type, handle slicing for the additive components
+        if isinstance(self, AdditiveBasis):
+            split_dict, start_slice = self._basis1._get_feature_slicing(
+                n_inputs[: len(self._basis1._n_basis_input)],
+                start_slice,
+                split_by_input=split_by_input,
+            )
+            sp2, start_slice = self._basis2._get_feature_slicing(
+                n_inputs[len(self._basis1._n_basis_input) :],
+                start_slice,
+                split_by_input=split_by_input,
+            )
+            split_dict = self._merge_slicing_dicts(split_dict, sp2)
+        else:
+            # Handle the default case for other basis types
+            split_dict, start_slice = self._get_default_slicing(
+                split_by_input, start_slice
+            )
+
+        return split_dict, start_slice
+
+    def _merge_slicing_dicts(self, dict1: dict, dict2: dict) -> dict:
+        """Merge two slicing dictionaries, handling key conflicts."""
+        for key, val in dict2.items():
+            if key in dict1:
+                new_key = self._generate_unique_key(dict1, key)
+                dict1[new_key] = val
+            else:
+                dict1[key] = val
+        return dict1
+
+    @staticmethod
+    def _generate_unique_key(existing_dict: dict, key: str) -> str:
+        """Generate a unique key if there is a conflict."""
+        extra = 1
+        new_key = f"{key}-{extra}"
+        while new_key in existing_dict:
+            extra += 1
+            new_key = f"{key}-{extra}"
+        return new_key
+
+    def _get_default_slicing(
+        self, split_by_input: bool, start_slice: int
+    ) -> Tuple[dict, int]:
+        """Handle default slicing logic."""
+        if split_by_input:
+            # should we remove this option?
+            if self._n_basis_input[0] == 1 or isinstance(self, MultiplicativeBasis):
```

Comment:
I would move the else in AdditiveBasis instead

Reply:
it is a condition that is matched by any non-composite basis receiving a single input or any multiplicative basis. I think the if close avoids redundancies, otherwise one would need to have an exact duplicate of the method in
 the MultiplicativeBasis,and the conv + eval mixin

Reply:
(I'll keep it the way it is now for the moment ) Also, this wasn't edited in this PR